### ISSUE TITLE
Finder: support CURRENT_AS_PATHNAME to improve efficency

### ISSUE THREAD

**Issue by staabm** [symfony/symfony#52065]
### Description

In https://github.com/phpstan/phpstan-src/pull/2682 we found out, that not using `SplFileInfo` in `RecursiveDirectoryIterator` can lead to a 20% improvement in memory and execution time.

since symfony explicitly disallows `CURRENT_AS_PATHNAME` in its `RecursiveDirectoryIterator` implementation

https://github.com/symfony/symfony/blob/7074da9b08d32182c9d079857cfde82f1014beb4/src/Symfony/Component/Finder/Iterator/RecursiveDirectoryIterator.php#L39-L41

I was wondering, whether it is acceptable to widen this constraint a bit or whether its a hard fact that symfony will not support this in the finder for the foreseeable future?

to sum up: I think for some use-cases it would be great to be able to make symfony/finder not use SplFileInfo, as it is more efficient.

**Comment by staabm** [comment]
would it be acceptable if I add support for the most used, but not all FilterIterators?

**Comment by staabm** [comment]
while working thru the problem again and doing some blackfire profiles - and also measures without blackfire - I conclude there is no performance opportunity.

when running on plain php there is nearly no difference between symfony/finder and plain spl iterators.
[I wrote the case down](https://github.com/staabm/hungry-spl-file-info/tree/main) and reported it to blackfire.
